x-pack/plugins/reporting/server/routes/lib/get_document_payload.ts

```diff
@@ -18,42 +21,44 @@ function getDocumentPayloadFn(server) {
     }
   }
 
-  function getCompleted(output, jobType, title) {
-    const exportType = exportTypesRegistry.get(item => item.jobType === jobType);
+  function getCompleted(output: any, jobType: string, title: any) {
+    const exportType = exportTypesRegistry.get((item: any) => item.jobType === jobType);
+    const filename = `${title || 'report'}.${exportType.jobContentExtension}`;
+
     return {
       statusCode: 200,
       content: encodeContent(output.content, exportType),
       contentType: output.content_type,
       headers: {
-        'Content-Disposition': `inline; filename="${title || 'report'}.${exportType.jobContentExtension}"`
-      }
+        'Content-Disposition': contentDisposition(filename, { type: 'inline' }),
+      },
     };
   }
 
-  function getFailure(output) {
+  function getFailure(output: any) {
     return {
       statusCode: 500,
       content: {
         message: 'Reporting generation failed',
-        reason: output.content
+        reason: output.content,
       },
-      contentType: 'application/json'
+      contentType: 'application/json',
     };
   }
 
-  function getIncomplete(status) {
+  function getIncomplete(status: any) {
     return {
       statusCode: 503,
       content: status,
       contentType: 'application/json',
       headers: {
-        'retry-after': 30
-      }
+        'retry-after': 30,
+      },
     };
   }
 
-  return function getDocumentPayload(doc) {
-    const { status, output, jobtype: jobType, payload: { title } = {} } = doc._source;
+  return function getDocumentPayload(doc: any) {
```

Comment:
After https://github.com/elastic/kibana/pull/31188 goes in, we can replace the `any` with a type :D